src/team_lib/mhba/xccl_mhba_lib.c

```diff
@@ -424,13 +424,13 @@ xccl_mhba_team_create_post(xccl_tl_context_t *context,
     calc_block_size(mhba_team);
     if (XCCL_MHBA_IS_ASR(mhba_team)) {
 
-        xccl_status_t status = xccl_mhba_init_umr(ctx);
+        xccl_status_t status = xccl_mhba_init_umr(ctx, &mhba_team->node);
         if (status!=XCCL_OK){
             xccl_mhba_error("Failed to init UMR");
             goto fail;
         }
 
-        mhba_team->net.cq = ibv_create_cq(ctx->ib_ctx, ctx->cfg.asr_cq_size, NULL, NULL, 0);
+        mhba_team->net.cq = ibv_create_cq(mhba_team->node.shared_ctx, ctx->cfg.asr_cq_size, NULL, NULL, 0);
```

Comment:
i remember we discussed that we don't need shared_ctx here. It is ASR only code. It can use regular ctx/pd.

Reply:
agreed to leave as is for now